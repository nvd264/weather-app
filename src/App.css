.App {
  margin: 20px auto;
}

.app-title {
  text-align: center;
}

.weather-row {  
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin-top: 60px;
  text-align: center;
}

.weather-row > div {
  flex-basis: 25%;
  flex-grow: 0;
}

.card-title {
  font-size: 24px;
}

.card-text {
  margin-top: 8px;
}

.search-box {
  position: relative;
  max-width: 320px;
}

.search-box > span {
  position: absolute;
  right: 4px;
  top: 0;
  line-height: 38px;
  color: #495057;
}

