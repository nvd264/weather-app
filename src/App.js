import PropTypes from 'prop-types';
import './App.css';

import SearchCity from './components/SearchCity';
import WeatherDay from './components/WeatherDay';
import Loading from './components/Loading';

function App({ loading, weatherList, getWeatherInfo }) {
  return (
    <div className="App">
      <div className="container">
        <h2 className="app-title">Weather App</h2>
        <SearchCity onSearch={getWeatherInfo} />
        {loading ? <Loading /> : (
          <div className="weather-row">
            {Object.keys(weatherList || {}).map(key =>
              <WeatherDay
                key={key}
                label={key}
                data={weatherList[key]}
              />
            )}
          </div>
        )}
        {!loading && !weatherList && <p>No data found</p>}
      </div>
    </div>
  );
}

App.propTypes = {
  weatherList: PropTypes.shape({}),
  loading: PropTypes.bool,
  getWeatherInfo: PropTypes.func.isRequired,
}

App.defaultProps = {
  weatherList: null,
  loading: false,
}

export default App;
