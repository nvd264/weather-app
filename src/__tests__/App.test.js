import { render, screen } from '@testing-library/react';
import App from '../App'; 

describe('App Component', () => {
  let props;
  
  beforeEach(() => {
    props = {
      weatherList: null,
      loading: false,
      getWeatherInfo: jest.fn(),
    };
  });
  
  test('Should match the snapshot on init', () => {
    const { container } = render(<App {...props} />);
    expect(container).toMatchSnapshot();
  });
  
  test('Should show loading', () => {
    props.loading = true;
    render(<App {...props} />);
    expect(screen.getByText(/Loading/)).toBeInTheDocument();
  });
  
  test('Should match the snapshot with weather info', () => {
    props.weatherList = {
      Thursday: {
        min: 1,
        max: 2,
        icon: 'test1',
        dateStr: '2021-1-21'
      },
      Friday: {
        min: 3,
        max: 4,
        icon: 'test2',
        dateStr: '2021-1-22'
      },
      Saturday: {
        min: 5,
        max: 6,
        icon: 'test3',
        dateStr: '2021-1-23'
      },
    };
    
    const { container } = render(<App {...props} />);
    expect(container).toMatchSnapshot();
  });
});


