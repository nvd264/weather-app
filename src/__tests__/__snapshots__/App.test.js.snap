// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should match the snapshot on init 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <h2
        class="app-title"
      >
        Weather App
      </h2>
      <div
        class="search-box"
      >
        <input
          aria-label="search-input"
          class="form-control"
          id="cityName"
          placeholder="City name..."
          type="text"
          value=""
        />
        
      </div>
      <div
        class="weather-row"
      />
      <p>
        No data found
      </p>
    </div>
  </div>
</div>
`;

exports[`App Component Should match the snapshot with weather info 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <h2
        class="app-title"
      >
        Weather App
      </h2>
      <div
        class="search-box"
      >
        <input
          aria-label="search-input"
          class="form-control"
          id="cityName"
          placeholder="City name..."
          type="text"
          value=""
        />
        
      </div>
      <div
        class="weather-row"
      >
        <div
          class="card-deck"
        >
          <div
            class="card bg-default"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Thursday
              </h5>
              <h6>
                2021-1-21
              </h6>
              <img
                alt="Weather on 2021-1-21"
                src="http://openweathermap.org/img/wn/test1@2x.png"
              />
              <p
                class="card-text"
              >
                1
                ° - 
                2
                °
              </p>
            </div>
          </div>
        </div>
        <div
          class="card-deck"
        >
          <div
            class="card bg-default"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Friday
              </h5>
              <h6>
                2021-1-22
              </h6>
              <img
                alt="Weather on 2021-1-22"
                src="http://openweathermap.org/img/wn/test2@2x.png"
              />
              <p
                class="card-text"
              >
                3
                ° - 
                4
                °
              </p>
            </div>
          </div>
        </div>
        <div
          class="card-deck"
        >
          <div
            class="card bg-default"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Saturday
              </h5>
              <h6>
                2021-1-23
              </h6>
              <img
                alt="Weather on 2021-1-23"
                src="http://openweathermap.org/img/wn/test3@2x.png"
              />
              <p
                class="card-text"
              >
                5
                ° - 
                6
                °
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
